test/role.test.js

```diff
@@ -208,4 +210,64 @@ describe('role model', function() {
 
   });
 
+  describe('listByPrincipalType', function() {
+    var sandbox;
+
+    beforeEach(function() {
+      sandbox = sinon.sandbox.create();
+    });
+
+    afterEach(function() {
+      sandbox.restore();
+    });
+
+    it('should fetch all models assigned to the role', function(done) {
+      var principalTypesToModels = {};
+      var runs = 0;
+      var mappings;
+
+      principalTypesToModels[RoleMapping.USER] = User;
+      principalTypesToModels[RoleMapping.APPLICATION] = Application;
+      principalTypesToModels[RoleMapping.ROLE] = Role;
+
+      mappings = Object.keys(principalTypesToModels);
+
+      mappings.forEach(function(principalType) {
+        var Model = principalTypesToModels[principalType];
+        Model.create({name:'test', email:'x@y.com', password: 'foobar'}, function(err, model) {
+          Role.create({name:'testRole'}, function(err, role) {
+            role.principals.create({principalType: principalType, principalId: model.id}, function(err, p) {
+              var pluralName = Model.pluralModelName.toLowerCase();
+              role[pluralName](function(err, models) {
+                assert(!err);
+                assert.equal(models.length, 1);
+                if (++runs === mappings.length) {
+                  done();
+                }
+              });
+            });
+          });
+        });
+      });
+    });
+
+    it('should apply query', function(done) {
+      User.create({name: 'Raymond', email: 'x@y.com', password: 'foobar'}, function(err, user) {
+        Role.create({name: 'userRole'}, function(err, role) {
+          role.principals.create({principalType: RoleMapping.USER, principalId: user.id}, function(err, p) {
+            var query = {fields:['id', 'name']};
+            sandbox.spy(User, 'find');
```

Comment:
I'm not familiar with `sinon`... what is this `spy()` even used for?

I'd just get rid of it... I like the idea of spies / mocks / etc... but I think its something that should be added from the beginning not in a minor PR a year after the tests were written.

On the other hand... if you want to create a PR that adds sinon to all the tests that would be pretty cool.